### PR TITLE
Edit the OIDC and OAuth2 Client and Filters Reference Guide

### PR DESCRIPTION
**Pull request by rolfedh**
Purpose: Further tweaks. [Docs] Final edits, QE and publishing prep work for the OIDC and OAuth2 Client and Filters Reference Guide
References: [QDOCS-536](https://issues.redhat.com/browse/QDOCS-536)

### CODE REVIEW

**Review by sberyozkin — APPROVED** [review]

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 1]
```diff
@@ -107,9 +107,9 @@ quarkus.oidc-client.grant-options.password.username=alice
 quarkus.oidc-client.grant-options.password.password=alice
 ----
 
-It can be further customized using a `quarkus.oidc-client.grant-options.password` configuration prefix, similar to how the client credentials grant can be customized.
+It can be further customized by using a `quarkus.oidc-client.grant-options.password` configuration prefix, similar to how the client credentials grant can be customized.
 
-==== Other Grants
+==== Other grants
 
 `OidcClient` can also help acquire the tokens by using grants that require some extra input parameters that cannot be captured in the configuration. These grants are `refresh_token` (with the external refresh token), `authorization_code`, as well as two grants which can be used to exchange the current access token, `urn:ietf:params:oauth:grant-type:token-exchange` and `urn:ietf:params:oauth:grant-type:jwt-bearer`.
```
[the remark this answers is not in the document]
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:115]: @sheilamjones, which sentence do you refer to ? The one above does not have `for it to acquire`. `it` would be `OidcClient`

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 2]
```diff
@@ -30,11 +30,11 @@ Add the following dependency:
 </dependency>
 ----
 
-`quarkus-oidc-client` extension provides a reactive `io.quarkus.oidc.client.OidcClient` which can be used to acquire and refresh tokens using SmallRye Mutiny `Uni` and `Vert.x WebClient`.
+The `quarkus-oidc-client` extension provides a reactive `io.quarkus.oidc.client.OidcClient`, which can be used to acquire and refresh tokens using SmallRye Mutiny `Uni` and `Vert.x WebClient`.
```
[the remark this answers is not in the document]
Comment (sberyozkin) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:33]: @sheilamjones 

>Use Quarkus extensions to acquire and refresh access tokens from OIDC and OAuth 2.0 compliant authorization servers and propagate access tokens.

it reads good, thanks, I'd only drop `authorization` as it does not play a role in this context

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 3]
```diff
@@ -3,7 +3,7 @@ This guide is maintained in the main Quarkus repository
 and pull requests should be submitted there:
 https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 ////
-= OpenID Connect (OIDC) and OAuth2 Client and Filters Reference Guide
+= OpenID Connect (OIDC) and OAuth2 client and filters reference guide
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:6]: Removed. Thanks.

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (rolfedh) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:33]: Thanks. Sheila! Applied all the suggestions above.

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 4]
```diff
@@ -30,11 +30,11 @@ Add the following dependency:
 </dependency>
 ----
 
-`quarkus-oidc-client` extension provides a reactive `io.quarkus.oidc.client.OidcClient` which can be used to acquire and refresh tokens using SmallRye Mutiny `Uni` and `Vert.x WebClient`.
+The `quarkus-oidc-client` extension provides a reactive `io.quarkus.oidc.client.OidcClient`, which can be used to acquire and refresh tokens using SmallRye Mutiny `Uni` and `Vert.x WebClient`.
 
-`OidcClient` is initialized at the build time with the IDP token endpoint URL, which can be auto-discovered or manually configured, and uses this endpoint to acquire access tokens using the token grants such as `client_credentials` or `password` and refresh the tokens using a `refresh_token` grant.
+`OidcClient` is initialized at the build time with the IDP token endpoint URL, which can be auto-discovered or manually configured. `OidcClient` uses this endpoint to acquire access tokens by using token grants such as `client_credentials` or `password` and refresh the tokens by using a `refresh_token` grant.
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:35]: Updated. Thanks.

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 5]
```diff
@@ -107,9 +107,9 @@ quarkus.oidc-client.grant-options.password.username=alice
 quarkus.oidc-client.grant-options.password.password=alice
 ----
 
-It can be further customized using a `quarkus.oidc-client.grant-options.password` configuration prefix, similar to how the client credentials grant can be customized.
+It can be further customized by using a `quarkus.oidc-client.grant-options.password` configuration prefix, similar to how the client credentials grant can be customized.
 
-==== Other Grants
+==== Other grants
 
 `OidcClient` can also help acquire the tokens by using grants that require some extra input parameters that cannot be captured in the configuration. These grants are `refresh_token` (with the external refresh token), `authorization_code`, as well as two grants which can be used to exchange the current access token, `urn:ietf:params:oauth:grant-type:token-exchange` and `urn:ietf:params:oauth:grant-type:jwt-bearer`.
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:114]: Updated. Thanks. Crafty!

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc (hunk 1 above)
[the remark this answers is not in the document]
Comment (rolfedh) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:115]: @sberyozkin How does this sound to you?
```suggestion
If you post an existing refresh token to the current Quarkus endpoint to acquire an access token, you will need to use the `refresh_token` grant. This grant allows you to obtain a new set of tokens through an out-of-band refresh token.
```

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 6]
```diff
@@ -556,26 +556,26 @@ The `Tokens` producer will acquire and refresh the tokens, and the custom filter
...
 
 Note that some OpenID Connect Providers will not return a refresh token in a `client_credentials` grant response. For example, starting from Keycloak 12, a refresh token will not be returned by default for `client_credentials`. The providers might also restrict the number of times a refresh token can be used.
 
 [[revoke-access-tokens]]
-=== Revoking Access Tokens
+=== Revoking access tokens
 
 If your OpenId Connect provider, such as Keycloak, supports a token revocation endpoint, then `OidcClient#revokeAccessToken` can be used to revoke the current access token. The revocation endpoint URL will be discovered alongside the token request URI or can be configured with `quarkus.oidc-client.revoke-path`.
 
 You might want to have the access token revoked if using this token with a REST client fails with HTTP `401` or if the access token has already been used for a long time and you'd like to refresh it.
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:573]: Updated. Thanks.

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 7]
```diff
@@ -1039,7 +1039,7 @@ public interface ProtectedResourceService {
 
 Alternatively, `AccessTokenRequestFilter` can be registered automatically with all MP Rest or Jakarta REST clients if the `quarkus.oidc-token-propagation.register-filter` property is set to `true` and `quarkus.oidc-token-propagation.json-web-token` property is set to `false` (which is a default value).
 
-==== Exchange Token Before Propagation
+==== Exchange token before propagation
 
 If the current access token needs to be exchanged before propagation and you work with link:https://www.keycloak.org/docs/latest/securing_apps/#_token-exchange[Keycloak] or other OpenID Connect Provider which supports a link:https://tools.ietf.org/html/rfc8693[Token Exchange] token grant then you can configure `AccessTokenRequestFilter` like this:
```
[the remark this answers is not in the document]
Comment (rolfedh) [reply docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:1044]: Updated. Thanks.

File: docs/src/main/asciidoc/security-openid-connect-client-reference.adoc [hunk 8]
```diff
@@ -3,11 +3,13 @@ This guide is maintained in the main Quarkus repository
...
-= OpenID Connect (OIDC) and OAuth2 Client and Filters Reference Guide
+= OpenID Connect (OIDC) and OAuth2 client and filters
 include::_attributes.adoc[]
+:diataxis-type: reference
 :categories: security
 :topics: security,oidc,client
 :extensions: io.quarkus:quarkus-oidc-client
+:summary: Use Quarkus extensions to acquire and refresh access tokens from OIDC and OAuth 2.0 compliant servers and propagate access tokens.
 
 This reference guide explains how to use:
```
Comment (sheilamjones) [inline docs/src/main/asciidoc/security-openid-connect-client-reference.adoc:15]: @rolfedh: Thanks for considering the changes. Just one comment on the corresponding line 16. @sberyozkin advised that "authorization" be removed from "OIDC and OAuth 2.0 compliant authorization servers" in the newly added summary line. Should this update also be made on line 16 and for that line to be tweaked to the following:

 "- `quarkus-oidc-client`, `quarkus-oidc-client-reactive-filter` and `quarkus-oidc-client-filter` extensions to acquire and refresh access tokens from OIDC and OAuth 2.0 compliant servers such as link:https://www.keycloak.org[Keycloak]"

**Review by sheilamjones — APPROVED** [review]

**Comment by sberyozkin** [comment]
Thanks @rolfedh @sheilamjones

**Pull request merged**